### PR TITLE
Focal loss one-hot target

### PR DESCRIPTION
**Pull request by wyli**
enhance the focal loss to support both onehot and non-onehot ground truth

co-authored by @wentaozhu

### Status
**Ready**

### Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [x] New tests added to cover the changes
- [x] Docstrings/Documentation updated

### CODE REVIEW

File: tests/test_focal_loss.py [hunk 1]
```diff
@@ -104,17 +104,22 @@ def test_multi_class_seg_2d(self):
...
         pred_very_good = 1000 * F.one_hot(target, num_classes=num_classes).permute(0, 3, 1, 2).float()
-
         # initialize the mean dice loss
         loss = FocalLoss()
 
         # focal loss for pred_very_good should be close to 0
+        target_one_hot = F.one_hot(target, num_classes=num_classes).permute(0, 3, 1, 2)  # test one hot
         target = target.unsqueeze(1)  # shape (1, 1, H, W)
+
         focal_loss_good = float(loss.forward(pred_very_good, target).cpu())
```
Comment (Nic-Ma) [inline tests/test_focal_loss.py:114]: Suggest using the "\__call__()" API instead of `forward()`, it's the formal API in PyTorch nn.Module:
```py
focal_loss_good = float(loss(pred_very_good, target).cpu())
```

Comment (wyli) [reply tests/test_focal_loss.py:114]: agreed. not sure why @myron changed it to `.forward()` in a previous discussion https://github.com/Project-MONAI/MONAI/pull/487#discussion_r436186238

Comment (myron) [reply tests/test_focal_loss.py]: hmm, I also think we should just use __call__ api.  I used .forward in test_masked_dice_loss.py, just because test_dice_loss.py was using it.
